graph/graph.go

```diff
@@ -163,14 +163,11 @@ func (graph *Graph) Register(img *image.Image, layerData archive.ArchiveReader)
 	graph.imageMutex.Lock(img.ID)
 	defer graph.imageMutex.Unlock(img.ID)
 
-	defer func() {
+	var cleanup = func() {
 		// If any error occurs, remove the new dir from the driver.
 		// Don't check for errors since the dir might not have been created.
-		// FIXME: this leaves a possible race condition.
-		if err != nil {
```

Comment:
I don't actually think this is getting shadowed: http://play.golang.org/p/1aHCqY9FZc.

That doesn't necessarily mean I think we shouldn't make this change.